packages/-ember-data/tests/integration/store-test.js

```diff
@@ -165,7 +164,8 @@ module('integration/store - destroy', function (hooks) {
 
       throw new Error("We shouldn't be pushing data into the store when it is destroyed");
     };
-    let requestPromise = store.findRecord('car', '1');
+
+    store.findRecord('car', '1');
 
     await nextPromise;
```

Comment:
I thought I was reading this right but now I'm questioning myself.  I wouldn't think we would advance from this line because we have yet to `resolve` the promise.  I thought it would just hang here.

Reply:
It gets resolved on line `143` no?

Reply:
[Here](https://www.typescriptlang.org/play?#code/BQQwzgngdgxgBAMwK6wC4EsD2VgEo4DeAUHKXADYCmqcUlAHqgFxwhQQDcJZVNdA7gAUATpgC26MJS5lEKGBmyJMmPIW6zSAkeMmU4AXlqV+cHRKnBgwymEzkAbpXwGAfOs2e4MbHaoA6ckwAc2AAchgQcnIAIxAYAGsw3BkvWTpGQzgbPydUzwBfXH9UAAtKHDU3DzSyHyg-SkCQ8LKK5I1am1QkYShCVhYwkDC4AvzNIq5O2W7e-u1RC2lOgqJO3mzKAEcshBU8CZB+EHQ+E3M9CfrG5tCwgBN0B9YHhzYYSmSJjNRDzpu9iaQXuOXsTgeHVkNxoDxAqBAWWOp3OQiWV06yLOW22118QLuwDhCJSRCK-yAA) is a plan TS example.  We haven't resolved yet.  Clearly a gap in my knowledge somewhere 😆

Reply:
I missed that store.findRecord would batch multiple if necessary.  As a result, nextPromise is undefined here.